### DISCUSSION TITLE
Summary of Practice Domain Retrospective

### DISCUSSION THREAD

**Discussion by mikewolfd** [FearlessSolutions/engineering-practice-domains#14] [Practice Domain Feedback]
We've been advancing the practice domain concept since the beginning of February, and we conducted a 6-month retrospective to better understand how we can iterate on our engineering practice as a whole and better support everyone's needs through Practice Domains.

Here is the original [miro](https://miro.com/welcomeonboard/dm8wSzRaTlJvYjlqQzdwUk82ZUt5MVNKT2hhNnNlRE1odUxLWmtrUDNna2VQTXMxYjFUOHlIQ25jQkRia0wzOXwzMDc0NDU3MzU2NTA5NzUwMjY1fDI=?share_link_id=909557568415) retro board. We had approximately 14 people attend and engaged in active and productive discourse.

### Key Insights:

- **Positive Reception**: The GitHub repo, project board, and wiki have been well-received as valuable resources for the team, anticipated to become even more useful over time. "The paradox of needing/generating documentation" - JP
- **Low Meeting Attendance**: Concerns about billable hours contribute to low attendance. Team members feel guilty attending due to the formal nature of the meetings, contrasting with the expected casual 'water cooler' chats.
- **Desire for Structured Communication**: There is a strong call for more structured meetings, with clear agendas and advanced scheduling to accommodate different time zones and work schedules.

### Actionable Changes:

- **Reduce and Streamline Meetings**: Focus on reducing the number of meetings and channels, iterating their format based on feedback and effectiveness.
- **Split Meeting Focus**:
  - **Delivery/Project-Oriented**: Casual but specific open forums for problem-solving and showcasing project successes or related open-source contributions.
  - **Domain-Oriented**: Standup meetings to review the GitHub Project board, development, and document status, and explore ideas. More structured meetings with specific themes, examples, and detailed discussions with explicit value and defined outcome structures.
- **Enhanced Asynchronous Engagement**: Explore asynchronous methods for conducting meetings to include those who cannot attend in real-time. Consider scheduling polls to better align with everyone’s schedules.
- **Leverage GitHub Discussions**: Increase the use of GitHub Discussions to foster ongoing dialogue and decrease the need for synchronous meetings, allowing for more flexible participation.

### Future Steps:

- Investigate how other organizations manage asynchronous engagement effectively.
- Increase Slack activity to maintain engagement and encourage continuous communication about ongoing projects and discussions, balancing engagement with respect for everyone's notification preferences.
- Encourage self-driven contributions by asking team members to suggest topics or issues for domain-oriented discussions, fostering a more participative environment with sources of feedback.
- Address concerns about billable hours by clarifying the value and expectations of these meetings, altering how they are perceived and used, and hopefully, to be recognized as essential, not optional.

**TL;DR**: We're refining our practice domain meetings to be more structured and engaging, dividing them into project-oriented and domain-oriented categories. We'll leverage our successful tools like the GitHub repo, Project board, and discussions more actively and address concerns around billable hours to ensure these meetings are valuable without feeling burdensome. 

![Screenshot 2024-08-05 at 4 57 51 PM](https://github.com/user-attachments/assets/20d8f559-7a31-47e8-9826-ab3e058e613c)

_This was workshopped with chatgpt_

**Comment by fearlessrob** [comment]
To fill in some details regarding how we can use these Github discussions as an anchor/artifact of participation;

An initial question in a discussion board here is a "call to action" where the reader understands they can take a few minutes to read the question prompt and toss in an answer or reply without any further obligation.

CTA: The question of the week is "How has a FISMA, NIST 800 or other security policy impact your project in an unexpected way?" Go to our Github discussion here and post a few sentences about it right now to help us understand the space!"

Post this call-to-action vigorously and repeatedly in Slack channels and The Beef posts and wherever. Have a few ringers you've talked to in advance to populate the first few replies so nobody feels like they're first. If related discussions happen on Slack have someone summarize or link to them from a post.

Over a few days the discussion post will start to evolve into a shape and gain some value as a document or suggestions for agenda items. Particularly interesting responses can be a 2-3 minute verbal summary by the poster, followed by discussion. As interesting things get posted, highlight them in subsequent slack posts touting the discussion thread URL link.

Hold the in person meeting based on how the discussion thread evolved, schedule it around the most active participants of the thread but leave room for others who might want to join.

Record the meeting but only for the purpose of summary notes afterwards. Post the summary notes to the chat and encourage any discussions that _would have_ happened in a longer meeting to occur on the discussion thread as a post-meeting and continued async discussion.

The thread has its own searchable value in Github now. Use it as an artifact to build other things such as documentation, marketing, or business development artifacts, particularly parts of the discussion that tell a story about how Fearless solves customer problems.

Wash, rinse, repeat.

**Reply by averysmithproductions** [reply]
I like this suggestion
